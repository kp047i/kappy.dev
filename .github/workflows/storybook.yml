name: Storybook Tests

on:
  push:
  pull_request:

jobs:
  storybook:
    name: Storybook Tests
    runs-on: ubuntu-latest
    timeout-minutes: 60
    env:
      PLAYWRIGHT_BROWSERS_PATH: 0
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: 22.13.0
      - name: Enable Corepack
        run: corepack enable
      - name: Install pnpm
        run: |
          corepack prepare pnpm@8.15.5 --activate
          pnpm --version
          which pnpm
      - name: Get pnpm store directory
        shell: bash
        run: |
          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
      - name: Setup pnpm cache
        uses: actions/cache@v4
        with:
          path: ${{ env.STORE_PATH }}
          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm-store-
      - name: Install dependencies
        run: pnpm install --frozen-lockfile
      - name: Install Playwright browsers
        run: pnpm exec playwright install --with-deps chromium chromium-headless-shell
      - name: Build Storybook
        run: pnpm build-storybook --quiet
      - name: Storybook Vitest tests
        env:
          CI: true
        run: pnpm run test:storybook
