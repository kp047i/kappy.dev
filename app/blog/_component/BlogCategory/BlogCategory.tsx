import { Metadata } from "next";
import { FaFolderOpen } from "react-icons/fa6";
import { FaFolderClosed } from "react-icons/fa6";
import { CATEGORIES } from "../../_const/categories";
import Link from "next/link";

export function BlogCategory({ category }: { category: Metadata["category"] }) {
  return (
    <Link
      href={`/blog?category=${category}`}
      className="inline-flex items-center gap-2 space-x-0 text-primary-300 opacity-80 group"
    >
      {/* ホバー時に開いたフォルダを表示、それ以外は閉じたフォルダ */}
      <FaFolderClosed className="transform group-hover:hidden" />
      <FaFolderOpen className="hidden transform group-hover:block group-hover:m-0" />
      <span>{CATEGORIES.find((c) => c.key === category)?.label}</span>
    </Link>
  );
}
