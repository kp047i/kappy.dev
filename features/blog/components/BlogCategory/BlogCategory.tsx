import { Metadata } from "next";
import Link from "next/link";
import { FaFolderOpen } from "react-icons/fa6";
import { FaFolderClosed } from "react-icons/fa6";

import { CATEGORIES } from "../../const/categories";

export function BlogCategory({ category }: { category: Metadata["category"] }) {
  return (
    <Link
      href={`/blog?category=${category}`}
      className="group inline-flex items-center gap-2 text-primary-700 opacity-80 transition-colors hover:opacity-100 dark:text-primary-300"
    >
      {/* ホバー時に開いたフォルダを表示、それ以外は閉じたフォルダ */}
      <FaFolderClosed className="transform text-current group-hover:hidden" />
      <FaFolderOpen className="hidden transform text-current group-hover:block group-hover:m-0" />
      <span>{CATEGORIES.find((c) => c.key === category)?.label}</span>
    </Link>
  );
}
