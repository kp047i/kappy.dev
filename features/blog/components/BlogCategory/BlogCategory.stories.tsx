import { Meta, StoryObj } from "@storybook/nextjs";
import { expect, within } from "storybook/test";

import { BlogCategory } from "./BlogCategory";

const meta: Meta<typeof BlogCategory> = {
  component: BlogCategory,
};

export default meta;
type Story = StoryObj<typeof BlogCategory>;

const defaultArgs = {
  category: "tech",
};

export const Default: Story = {
  args: defaultArgs,
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);

    await expect(canvas.getByText("Tech")).toBeVisible();

    const link = canvas.getByText("Tech");
    const parent = link.parentElement;
    expect(parent).toHaveAttribute("href", "/blog?category=tech");
  },
};
