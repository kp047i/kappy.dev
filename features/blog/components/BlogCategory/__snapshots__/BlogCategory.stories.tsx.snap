// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog/components/BlogCategory Default play-test 1`] = `
<script nonce>
  ((e,i,s,u,m,a,l,h)=>{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=>{let k=y==="class",S=k&&a?m.map(f=>a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","light",null,["light","dark"],null,false,true)
</script>
<a class="group inline-flex items-center gap-2 text-primary-700 opacity-80 transition-colors hover:opacity-100 dark:text-primary-300"
   href="/blog?category=tech"
>
  <svg stroke="currentColor"
       fill="currentColor"
       stroke-width="0"
       viewbox="0 0 512 512"
       class="transform text-current group-hover:hidden"
       height="1em"
       width="1em"
       xmlns="http://www.w3.org/2000/svg"
  >
    <path d="M448 480L64 480c-35.3 0-64-28.7-64-64L0 192l512 0 0 224c0 35.3-28.7 64-64 64zm64-320L0 160 0 96C0 60.7 28.7 32 64 32l128 0c20.1 0 39.1 9.5 51.2 25.6l19.2 25.6c6 8.1 15.5 12.8 25.6 12.8l160 0c35.3 0 64 28.7 64 64z">
    </path>
  </svg>
  <svg stroke="currentColor"
       fill="currentColor"
       stroke-width="0"
       viewbox="0 0 576 512"
       class="hidden transform text-current group-hover:block group-hover:m-0"
       height="1em"
       width="1em"
       xmlns="http://www.w3.org/2000/svg"
  >
    <path d="M88.7 223.8L0 375.8 0 96C0 60.7 28.7 32 64 32l117.5 0c17 0 33.3 6.7 45.3 18.7l26.5 26.5c12 12 28.3 18.7 45.3 18.7L416 96c35.3 0 64 28.7 64 64l0 32-336 0c-22.8 0-43.8 12.1-55.3 31.8zm27.6 16.1C122.1 230 132.6 224 144 224l400 0c11.5 0 22 6.1 27.7 16.1s5.7 22.2-.1 32.1l-112 192C453.9 474 443.4 480 432 480L32 480c-11.5 0-22-6.1-27.7-16.1s-5.7-22.2 .1-32.1l112-192z">
    </path>
  </svg>
  <span>
    Tech
  </span>
</a>
`;
